server/index.js

```diff
@@ -9,11 +9,38 @@ const staticMiddleware = require('./static-middleware');
 const uploadsMiddleware = require('./uploads-middleware');
 const authorizationMiddleware = require('./authorization-middleware');
 const http = require('http');
-const socketio = require('socket.io');
+const { Server } = require('socket.io');
 
 const app = express();
-const server = http.createServer(app);
-const io = socketio(server);
+const httpServer = http.createServer(app);
+const io = new Server(httpServer);
+
+io.use((socket, next) => {
+  const accessToken = socket.handshake.auth.token;
+
+  if (!accessToken) {
+    throw new ClientError(401, 'authentication required');
+  }
+
+  const payload = jwt.verify(accessToken, process.env.TOKEN_SECRET);
+
+  socket.user = payload;
+  next();
+});
+
+io.on('connection', socket => {
+  // if (!socket.handshake.query.otherUserId) {
+  //   socket.disconnect();
+  //   return;
+  // }
```

Comment:
You should leave in this guard. If no `otherUserId` is provided, they're wasting a socket connection.